modules/core/04-channel/keeper/packet.go

```diff
@@ -132,6 +132,24 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected channel state to be one of [%s, %s], but got %s", types.OPEN, types.FLUSHING, channel.State)
 	}
 
+	// If counterpartyUpgrade is stored we need to ensure that the
+	// packet sequence is < counterparty next sequence send. This is a defensive
+	// check and if the counterparty is implemented correctly, this should never abort.
+	counterpartyUpgrade, found := k.GetCounterpartyUpgrade(ctx, packet.GetDestPort(), packet.GetDestChannel())
+	if found {
+		// only error if the counterparty next sequence send is set (> 0)
+		// this error should never be reached, as under normal circumstances the counterparty
+		// should not send any packets after the upgrade has been started.
+		counterpartyNextSequenceSend := counterpartyUpgrade.NextSequenceSend
+		if packet.GetSequence() >= counterpartyNextSequenceSend {
```

Comment:
The zero check in the spec is effectively checking if the counterparty upgrade is not set. Since we do this inside `if found` clause it is unnecessary here.

If the counterpartyUpgrade is set, the counterpartyNextSequenceSend must be set to a nonzero value